src/ops/controlflow.rs

```diff
@@ -169,12 +169,21 @@ impl OpTrait for BasicBlock {
         Some(EdgeKind::ControlFlow)
     }
 
-    fn signature(&self) -> FunctionType {
-        match self {
+    fn signature(&self) -> Option<FunctionType> {
```

Comment:
Hmmmm, so this is because we need to have a FunctionType in order to have a `Signature` carrying the input-extensions, right?

Is it time to decouple input-extensions from signature? (Another issue, not here.)

Reply:
yes exactly, I think `Signature` should be a) renamed, and b) hold an `Option<FunctionType>` ?

Reply:
I actually think two separate members on NodeType (`input_extensions: Option<ExtensionSet>, signature: Option<FunctionType>`) might be better, but if keeping them together then yes